cel/cos_tlv.go

```diff
@@ -37,3 +57,17 @@ func (c CosTlv) GenerateDigest(hashAlgo crypto.Hash) ([]byte, error) {
 	}
 	return hash.Sum(nil), nil
 }
+
+// NewCosTlvFromTlv constructs a CosTlv from a TLV. It will check for the correct COS event
+// type, and unmarshal the nested event.
+func NewCosTlvFromTlv(tlv TLV) (CosTlv, error) {
+	if tlv.Type != CosEventType {
+		return CosTlv{}, fmt.Errorf("not a COS event")
```

Comment:
would be helpful to see the seen type in the error: `return CosTlv{}, fmt.Errorf("TLV type %v not a COS event", tlv.Type)`

Reply:
done